### PR TITLE
Add test upgrading Protocol 20 to Protocol 21

### PR DESCRIPTION
**Pull request by 2opremio**
### What

Test upgrading protocol 20 to protocol 21

### Why

Make sure everything works

### Known limitations

N/A

### CODE REVIEW

File: cmd/soroban-rpc/internal/test/upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	newLedgerSequence := uint32(0)
+	for i := 0; i < 60; i++ {
+		time.Sleep(time.Second)
+		newLedgerSequence, err = db.NewLedgerEntryReader(rpcDB).GetLatestLedgerSequence(context.Background())
+		require.NoError(t, err)
+		if newLedgerSequence > initialLedgerSequence {
+			break
+		}
+	}
+	require.Greater(t, newLedgerSequence, initialLedgerSequence, "rpc didn't start ingesting protocol 21")
```
Comment (sreuland) [inline cmd/soroban-rpc/internal/test/upgrade_test.go:68]: could remove the `for loop` in favor of `require.Eventually(t, func() {...}, time.Minute, time.Second)`

File: cmd/soroban-rpc/internal/test/upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+		},
+		BaseFee: txnbuild.MinBaseFee,
+		Preconditions: txnbuild.Preconditions{
+			TimeBounds: txnbuild.NewInfiniteTimeout(),
+		},
+	})
+	tx, err = txnbuild.NewTransaction(params)
+	assert.NoError(t, err)
+
+	assert.NoError(t, err)
```
Comment (sreuland) [inline cmd/soroban-rpc/internal/test/upgrade_test.go:126]: duplicate assertion on ln 124 and 126. 
is it worthwhile to check return value/SorobanData on the tx result to confirm invoke host fn response?

**Review by sreuland — APPROVED** [review]

File: cmd/soroban-rpc/internal/test/upgrade_test.go (hunk 1 above)
Comment (2opremio) [reply cmd/soroban-rpc/internal/test/upgrade_test.go:68]: Nice, I will use this!

File: cmd/soroban-rpc/internal/test/upgrade_test.go (hunk 2 above)
Comment (2opremio) [reply cmd/soroban-rpc/internal/test/upgrade_test.go:126]: Good catch

File: cmd/soroban-rpc/internal/test/upgrade_test.go (hunk 1 above)
Comment (2opremio) [reply cmd/soroban-rpc/internal/test/upgrade_test.go:68]: We should probably use this in the state expiration tests as well

**Pull request merged**